operations/service_deploy_smoke_utils.go

```diff
@@ -178,31 +168,131 @@ func triggerRepotracker(username, key string, client *http.Client) error {
+// create versions for them. The particular versions that it creates for these
+// commits is not that important, only that they exist.
+func waitForRepotracker(projectName, username, key string, client *http.Client) error {
+	grip.Info("Waiting for repotracker to pick up new commits.")
 
-	const buildCheckAttempts = 30
-	for i := 0; i < buildCheckAttempts; i++ {
-		// Poll the app server until the builds exist and have tasks.
-		// This is implicitly assuming that the app server has proper
-		// integration with GitHub to pick up the latest commit and successfully
-		// creates versions, builds, and tasks based on that commit.
+	const repotrackerAttempts = 10
+	for i := 0; i < repotrackerAttempts; i++ {
+		time.Sleep(2 * time.Second)
+		respBody, err := makeSmokeRequest(username, key, http.MethodGet, client, fmt.Sprintf("/rest/v2/projects/%s/versions?limit=1", projectName))
+		if err != nil {
+			grip.Error(errors.Wrapf(err, "requesting latest version for project '%s'", projectName))
+			continue
+		}
+		if len(respBody) == 0 {
+			grip.Error(errors.Errorf("did not find any latest revisions yet for project '%s'", projectName))
+			continue
+		}
 
-		time.Sleep(10 * time.Second)
+		// The repotracker should pick up new commits and create versions;
+		// therefore, the version creation time should be just a few moments
+		// ago.
+		latestVersions := []model.APIVersion{}
+		if err := json.Unmarshal(respBody, &latestVersions); err != nil {
+			grip.Error(errors.Wrap(err, "reading version create time from response body"))
+			continue
+		}
+		if len(latestVersions) == 0 {
+			grip.Error(errors.Errorf("listing latest versions for project '%s' yielded no results", projectName))
+			continue
+		}
+
+		latestVersion := latestVersions[0]
+		latestVersionID := utility.FromStringPtr(latestVersion.Id)
+		if createTime := utility.FromTimePtr(latestVersion.CreateTime); time.Since(createTime) > 365*24*time.Hour {
+			grip.Infof("Found latest version '%s' for project '%s', but it was created at %s, which was a long time ago, waiting for repotracker to pick up newer commit.", latestVersionID, projectName, createTime)
+			continue
+		}
 
-		// The latest GitHub commit must match the one that the repotracker
-		// picks up.
-		latest, err := getLatestGithubCommit()
+		grip.Infof("Repotracker successfully picked up a new commit '%s' and created version '%s'.", utility.FromStringPtr(latestVersion.Revision), latestVersionID)
+		return nil
+	}
+
+	return errors.Errorf("timed out waiting for repotracker to pick up new commits and create versions after %d attempts", repotrackerAttempts)
+}
+
+// submitSmokeTestPatch submits a manual patch to the app server to run the
+// smoke test and selects tasks in the given build variant.
+// Note that this requires using the CLI because there's currently no way to
+// create a patch from the REST API.
+func submitSmokeTestPatch(cliPath, cliConfigPath, bvToRun string) error {
```

Comment:
Should we maybe eventually create a cli command option like --smoke-test for manual testing purposes?

Reply:
I don't think this is possible because the main difficult with running the smoke test locally is populating secrets.

Reply:
What I meant is to create an "alias" that allows people to easily submit a smoke test patch to evergreen.

Reply:
Done.